packages/core/src/sync-realtime/service.ts

```diff
@@ -704,13 +709,17 @@ export class RealtimeSyncService extends Emittery<RealtimeSyncEvents> {
         lightMatchedLog.blockHash !== localLogs[i].blockHash ||
         lightMatchedLog.logIndex !== localLogs[i].logIndex
       ) {
-        await handleReorg(i);
+        await handleReorg(localLogs[i].blockNumber - 1);
```

Comment:
Maybe add a debug/trace level log here so we can confirm that we're hitting this branch, and the updated branch below?